### PR TITLE
Use source_policy_documents

### PR DESCRIPTION
**Pull request by bwmetcalf**
## what
* Use source_policy_documents instead of deprecated source_json

## why
* Eliminate warning

https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/iam_policy_document#source_json

### CODE REVIEW

**Comment by jamengual** [comment]
is the warning on TF 1.x ? the docs you posted are for `aws` provider 4.x and this module calls for 3.x.

**Comment by bwmetcalf** [comment]
> is the warning on TF 1.x ? the docs you posted are for `aws` provider 4.x and this module calls for 3.x.

Yes.  This is with aws provider v4 and I'm using tf 1.1.3.

**Comment by osterman** [comment]
>  This is with aws provider v4 and I'm using tf 1.1.3. v3 also uses source_policy_documents:

Sounds like we should upgrade the minimum version for pinning in that case

**Comment by jamengual** [comment]
the test are failing in the curl request of the zip, I already verified that the URL is correct and the file exist, it is weird.

**Comment by bwmetcalf** [comment]
> > This is with aws provider v4 and I'm using tf 1.1.3. v3 also uses source_policy_documents:
> 
> Sounds like we should upgrade the minimum version for pinning in that case

v4 deprecated source_json and I see from the docs that source_policy_documents is available in v3.30 and possibly earlier.

**Comment by woz5999** [comment]
This seems to be failing for reasons unrelated to the changes in the PR

```
TestExamplesComplete 2022-04-27T18:35:10Z retry.go:99: Returning due to fatal error: FatalError{Underlying: error while running command: exit status 1; 
Error: rpc error: code = Unavailable desc = transport is closing
```

**Comment by bwmetcalf** [comment]
Can we get this merged?  Anything else needed?  Thanks.

**Comment by Gowiem** [comment]
@bwmetcalf can you put your https://github.com/cloudposse/terraform-external-module-artifact branch up on PR and then add me as a reviewer? I'll be sure to prioritize and help out as I can see you've been waiting and working to get this merged.

**Comment by Gowiem** [comment]
@bwmetcalf both bats and terratest tests are failing due to the custom module, so let's work on upstreaming those changes first and then we can circle back onto this PR. Feel free to ping me in SweetOps if you need anything or when you're ready for a review. Thanks! 👍

**Comment by bwmetcalf** [comment]
@Gowiem Thanks.  My changes to `terraform-external-module-artifact` are a duplicate of this PR that is already open: 
 https://github.com/cloudposse/terraform-external-module-artifact/pull/19.  If we can merge that I'll update my changes here.

**Comment by Gowiem** [comment]
@bwmetcalf mind putting a new PR up for the change requested by @nitrocode [here](https://github.com/cloudposse/terraform-external-module-artifact/pull/19#discussion_r879720182)?

**Comment by bwmetcalf** [comment]
https://github.com/cloudposse/terraform-external-module-artifact/pull/21

File: lambda-log.tf [hunk 1]
```diff
@@ -34,9 +34,10 @@ module "forwarder_log_s3_label" {
 }
 
 module "forwarder_log_artifact" {
-  count   = local.lambda_enabled && var.forwarder_log_enabled ? 1 : 0
-  source  = "cloudposse/module-artifact/external"
-  version = "0.7.1"
+  count = local.lambda_enabled && var.forwarder_log_enabled ? 1 : 0
+  #  source  = "cloudposse/module-artifact/external"
+  #  version = "0.7.1"
+  source = "git::https://github.com/bwmetcalf/terraform-external-module-artifact.git?ref=remove-template-provider"
```
Comment (nitrocode) [inline lambda-log.tf:40]: We would need to use the cloud posse module here

**Comment by nitrocode** [comment]
Thanks for the original contribution. I think this can now be closed by https://github.com/cloudposse/terraform-aws-datadog-lambda-forwarder/pull/44.

**Pull request closed without merging**